### PR TITLE
Add full arity support to property methods

### PR DESCRIPTION
**Pull request by eparejatobes**

### CODE REVIEW

File: src/main/java/com/bio4j/angulillos/AnyElementType.java [hunk 1]
```diff
@@ -59,8 +59,11 @@
   /* The graph in which this element lives. */
   default G graph() { return type().graph(); }
 
-  /* The `get` method lets you get the value of a `property` which this element has. For that, you pass as an argument the [property](Property.java.md). Note that the type bounds only allow properties of this element. */
-  <X> X get(Property<FT,X> property);
+  /* The `get` method lets you get the value of a `property` which this element has. For that, you pass as an argument the [property](Property.java.md). Note that the type bounds only allow properties of this element, which are declared to be always defined. */
+  <X, P extends Property<FT,X> & Arity.ToAtLeastOne> X get(P property);
```
Comment (laughedelic) [inline src/main/java/com/bio4j/angulillos/AnyElementType.java]: I still think that it should be `ToOne` 😐

Comment (eparejatobes) [inline src/main/java/com/bio4j/angulillos/AnyElementType.java]: let's make Property extend `ToAtMostOne` if you're so inclined

**Comment by eparejatobes** [comment]
@laughedelic LGTM

File: src/main/java/com/bio4j/angulillos/AnyElementType.java (hunk 1 above)
Comment (laughedelic) [inline src/main/java/com/bio4j/angulillos/AnyElementType.java]: Why that?

**Comment by laughedelic** [comment]
> let's make Property extend ToAtMostOne if you're so inclined

OK. Also before merging, something should be done with the property classes in `TypedGraph`

**Comment by eparejatobes** [comment]
> OK. Also before merging, something should be done with the property classes in TypedGraph

Let's drop it, and use arities directly. Btw, I had a hierarchy of arity interfaces before that was used for marking that ToOne is a refinement of AtLeastOne and AtMostOne, for example.

**Comment by eparejatobes** [comment]
> Let's drop it, and use arities directly. Btw, I had a hierarchy of arity interfaces before that was used for marking that ToOne is a refinement of AtLeastOne and AtMostOne, for example.

@laughedelic please check this

**Comment by laughedelic** [comment]
> Let's drop it, and use arities directly

OK, if this is done:

> let's make Property extend ToAtMostOne if you're so inclined

with this:

>  I had a hierarchy of arity interfaces before that was used for marking that ToOne is a refinement of AtLeastOne and AtMostOne, for example

I don't remember whether it was there before, but yes, I suggested the same thing this morning.

**Comment by eparejatobes** [comment]
@laughedelic fine, done. On a different note, I don't know if you're aware of being lured into treating properties as edges :stuck_out_tongue:

**Comment by eparejatobes** [comment]
LGTM

**Comment by eparejatobes** [comment]
now for real LGTM

**Comment by laughedelic** [comment]
LGTM 👍

**Pull request merged**